server/src/main/java/org/elasticsearch/search/lookup/LeafStoredFieldsLookup.java

```diff
@@ -28,18 +28,19 @@
 public class LeafStoredFieldsLookup implements Map<Object, FieldLookup> {
 
     private final Function<String, MappedFieldType> fieldTypeLookup;
-    private final CheckedBiConsumer<Integer, StoredFieldVisitor, IOException> reader;
+    private final CheckedSupplier<StoredFields, IOException> storedFieldsSupplier;
 
     private int docId = -1;
+    private StoredFields storedFields = null;
```

Comment:
Can this be a `CachedSupplier` I suppose we would need a `CheckedCachedSupplier`, but it seems like that would be better as it would be `final` and threadsafe.

Reply:
LeafStoredFieldsLookup isn't thread-safe, and will need a lot of reworking before it can be, so I'm not sure it's worth adding a `CheckedCachedSupplier` here yet?